### ISSUE TITLE
Concerns about whitespace in the new #line grammar rules

### ISSUE THREAD

**Issue by RexJaeschke** [dotnet/docs#29693]
I'm writing a formal spec for this feature for the C# standards committee, and have some comments re the proposed new grammar rules:

```ANTLR
line_indicator
    : '(' decimal_digit+ ',' decimal_digit+ ')' '-' '(' decimal_digit+ ',' decimal_digit+ ')' decimal_digit+ whitespace file_name
    | '(' decimal_digit+ ',' decimal_digit+ ')' '-' '(' decimal_digit+ ',' decimal_digit+ ')' file_name
    ...
    ;
```

As you can see, *whitespace* is **required** in one place and **disallowed** between all other tokens. This doesn't look right!

I'm using Visual Studio Community 2022 (64-bit) Version 17.2.0 for my tests.

The compiler accepts the following source lines, without error:

```csharp
#line (  2  ,  3  )  -  (  4  ,  5  )  "f"
#line (  2  ,  3  )  -  (  4  ,  5  )  6  "f"
```

which suggests that an arbitrary amount of white space is permitted between any two tokens

The compiler also accepts

```csharp
#line (2,3)-(4,5)6"f"
```

where there is no white space between the character-offset number, 6, and the filename, contrary to the proposed grammar.

Note that like this proposal, the long-time directive format `#line 123 "f"` *does* require white space between the line number and filename, which is consistent. However, the compiler relaxes that in the test above, which means it's an extension or a bug.

Assuming that the one required white space place is correct, and that an arbitrary amount of white space is allowed between tokens, the new grammar rules should be

```ANTLR
line_indicator
    : '(' whitespace? decimal_digit+ whitespace? ',' whitespace? decimal_digit+ whitespace? ')' 
      whitespace? '-' whitespace?
      '('  whitespace? decimal_digit+ whitespace? ',' whitespace? decimal_digit+ whitespace? ')'
      whitespace? decimal_digit+ whitespace file_name
    | '(' whitespace? decimal_digit+ whitespace? ',' whitespace? decimal_digit+ whitespace? ')'
      whitespace? '-' whitespace?
      '('  whitespace? decimal_digit+ whitespace? ',' whitespace? decimal_digit+ whitespace? ')' whitespace? file_name
    ...
    ;
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 19676f37-72c5-ace5-f1e0-f2693c0ffe46
* Version Independent ID: ee32bc8f-4a21-dba1-5ab4-38852321190e
* Content: [Enhanced #line directives - C# 10.0 draft specifications](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/enhanced-line-directives#examples)
* Content Source: [proposals/csharp-10.0/enhanced-line-directives.md](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/enhanced-line-directives.md)
* Product: **dotnet-csharp**
* Technology: **csharp-spec**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **wiwagn**

**Comment by cston** [comment]
cc @tmat

**Comment by cston** [comment]
I've discussed this with @tmat. 

The parser should require spaces:
- before the first `(`
- before the character offset
- before the file name.

Spaces should be optional elsewhere.

I've logged https://github.com/dotnet/roslyn/issues/61663.

**Comment by BillWagner** [comment]
Fixed in dotnet/csharplang#6195
